auslib/blobs/gmp.py

```diff
@@ -61,6 +61,15 @@ def getInnerXML(self, updateQuery, update_type, whitelistedDomains, specialForce
 
         return vendorXML
 
+    def validate(self, *args, **kwargs):
+        Blob.validate(self, *args, **kwargs)
+        hashes = dict(SHA1=40, SHA256=64, SHA512=128)
+        required_length = hashes[self["hashFunction"]]
```

Comment:
We don't enforce that anywhere, so that's not a safe assumption to make. Using .upper() is a good idea.